jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPart.java

```diff
@@ -1392,6 +1397,7 @@ private boolean parseContent(Content.Chunk chunk)
             partialBoundaryMatch = boundaryFinder.endsWith(buffer);
             if (partialBoundaryMatch > 0)
             {
+                notifyCRContent();
```

Comment:
If I'm reading this correctly, this is notifying content of any hanging CR found in front of a partial match of a boundary.    But this is contrary to the comment below that says it is looking for `\n--Boundary`, so if found the CR should not be emitted as content... unless the next chunk happens not to contain the rest of the boundary and the partial boundary itself is content.       

I don't think we can emit the CR here until we know if we have a full boundary or not (in the next chunk).   

I think we need a state that remembers if there is a hanging CR or not .. which is kind of the existing `crContent` boolean?  

The problem might be in the code below that does `buffer.get(sliceLimit - 1) == '\r'` to set `crContent`, which wont work of the CR was in a previous chunk.    Perhaps the `crFlag` above is the same thing as `crContent` and we should set it above?

@lachlan-roberts can you help with this? I think this is your code???

Reply:
Note: this specific change was split out into PR #11409, as it's not related to the LEGACY stuff that this PR is doing.

Reply:
No longer relevant to this PR, this was handled separately in PR #11409